paimon-spark/paimon-spark-common/src/main/scala/org/apache/paimon/spark/aggregate/LocalAggregator.scala

```diff
@@ -0,0 +1,109 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.paimon.spark.aggregate
+
+import org.apache.paimon.manifest.PartitionEntry
+import org.apache.paimon.table.{DataTable, Table}
+
+import org.apache.spark.sql.catalyst.InternalRow
+import org.apache.spark.sql.connector.expressions.aggregate.{AggregateFunc, Aggregation, CountStar}
+import org.apache.spark.sql.types.{DataType, LongType, StructField, StructType}
+
+class LocalAggregator(table: Table) {
+  private var aggFuncEvaluator: Seq[AggFuncEvaluator[_]] = _
+
+  private def initialize(aggregation: Aggregation): Unit = {
+    aggFuncEvaluator = aggregation.aggregateExpressions().map {
+      case _: CountStar => new CountStarEvaluator()
+      case _ => throw new UnsupportedOperationException()
+    }
+  }
+
+  private def supportAggregateFunction(func: AggregateFunc): Boolean = {
+    func match {
+      case _: CountStar => true
+      case _ => false
+    }
+  }
+
+  def supportAggregation(aggregation: Aggregation): Boolean = {
```

Comment:
There is an `initialize` in this `supportAggregation`. It looks like not a `support`. Can you rename this to `pushAggregation`?

Reply:
+1, changed to `pushAggregation`